core/src/banking_stage/latest_unprocessed_votes.rs

```diff
@@ -207,26 +209,41 @@ impl LatestUnprocessedVotes {
     pub fn update_latest_vote(
         &self,
         vote: LatestValidatorVotePacket,
+        should_replenish_taken_votes: bool,
     ) -> Option<LatestValidatorVotePacket> {
         let pubkey = vote.pubkey();
         let slot = vote.slot();
         let timestamp = vote.timestamp();
 
+        // Allow votes for later slots or the same slot with later timestamp (refreshed votes)
+        // We directly compare as options to prioritize votes for same slot with timestamp as
+        // Some > None
+        let allow_update = |latest_vote: &LatestValidatorVotePacket| -> bool {
+            match slot.cmp(&latest_vote.slot()) {
+                cmp::Ordering::Less => return false,
+                cmp::Ordering::Greater => return true,
+                cmp::Ordering::Equal => {}
+            };
+
+            // Slots are equal, now check timestamp
+            match timestamp.cmp(&latest_vote.timestamp()) {
+                cmp::Ordering::Less => return false,
+                cmp::Ordering::Greater => return true,
+                cmp::Ordering::Equal => {}
+            };
```

Comment:
nit: can be combined with `match (slot, timestamp).cmp((...)) { ... }` maybe?

Reply:
That's neat. I'm satisfied with the less elegant way though :)